### PR TITLE
Some fixes for StaticModuleRecord tests

### PR DESCRIPTION
**Pull request by michaelfig**
I was able to fix a few of these tests, and improve the clarity of the Babel transform.

Stacked on 709-export-name-as-default.

### CODE REVIEW

File: packages/static-module-record/src/babelPlugin.js [hunk 1]
```diff
@@ -61,65 +91,133 @@ function makeModulePlugins(options) {
       allowedHiddens.add(id);
     };
 
-    const rewriteVars = (vids, isConst, needsHoisting) =>
-      vids.reduce((prior, id) => {
+    /**
```
Comment (kriskowal) [inline packages/static-module-record/src/babelPlugin.js:94]: I posted a follow-up that refactors `markExported` to require less logic by breaking this into `markLiveExport`, `markFixedExport`, and `markExport` (which infers which of the other to use only on whether the named export is initialized more than once). I also comments with my understanding from a close review.

**Review by kriskowal — APPROVED** [review]
The change looks good. I posted follow-up commits to add what I learned from the review and a couple drive-by refactors to simplify where I think it’s possible to simplify. There remain a couple places I don’t understand well enough to add good comments yet, noted in this review.

One of them changes the source type to be an invariant instead of a parameter. I think that’s valid, but I left that as an individual commit to make it easy to back out.

I also changed `reduce` to `flatMap` or just a loop. I like `reduce` like the worst of us, but I think it’s okay to lean on `flatMap` now.

File: packages/static-module-record/src/babelPlugin.js [hunk 2]
```diff
@@ -281,6 +300,16 @@ function makeModulePlugins(options) {
               expr.async,
             );
           }
+
+          if (decl.id) {
```
[the remark this answers is not in the document]
Comment (michaelfig) [reply packages/static-module-record/src/babelPlugin.js:383]: `id` is the declared identifier, used in `function id(...) { }` (also `async function id()` and `async function* id()`) or `class id { }`.

In that case, we don't want to use the tricky `const { default: $c_default } = { default: (XXX) };` assign-the-name-default-while-destructuring.

File: packages/static-module-record/src/babelPlugin.js (hunk 1 above)
Comment (michaelfig) [reply packages/static-module-record/src/babelPlugin.js:94]: That sounds great!  I'll have a look.

**Review by michaelfig — COMMENTED** [review]
LGTM!  Thanks for doing this.

(I can't approve since I'm the PR author.  But you can, and merge if you'd like.)

**Pull request merged**